src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt

```diff
@@ -83,99 +77,105 @@ class SpyFakku : HttpSource() {
+            thumbnail_url = doc.selectFirst("img[src*='/cover']")!!.absUrl("src")
+            description = buildString {
+                doc.select("a[href*='circle:']").emptyOrNull()?.joinToString { it.ownText() }?.let {
+                    append("Circles: ", it, "\n")
+                }
+                doc.select("a[href*='publisher:']").emptyOrNull()?.joinToString { it.ownText() }?.let {
+                    append("Publishers: ", it, "\n")
+                }
+                doc.select("a[href*='magazine:']").emptyOrNull()?.joinToString { it.ownText() }?.let {
+                    append("Magazines: ", it, "\n")
+                }
+                doc.select("a[href*='event:']").emptyOrNull()?.joinToString { it.ownText() }?.let {
+                    append("Events: ", it, "\n")
+                }
+                doc.select("a[href*='parody:']").emptyOrNull()?.joinToString { it.ownText() }?.let {
+                    append("Parodies: ", it, "\n\n")
+                }
+                append(
+                    "Created At: ",
+                    dateReformat.format(
+                        dateFormat.parse(doc.selectFirst("p:containsOwn(Released)")!!.parent()!!.selectFirst(".text-sm")!!.ownText())!!.time,
+                    ),
+                    "\n",
+                )
+                append("Pages: ", doc.selectFirst("p:containsOwn(pages)")!!.ownText(), "\n")
             }
-            append(
-                "Created At: ",
-                dateFormat.format(
-                    Date(createdAt * 1000),
-                ),
-                "\n",
-            )
-            append("Pages: ", pages, "\n")
+            status = SManga.COMPLETED
+            update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+            initialized = true
         }
-        status = SManga.COMPLETED
-        update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
-        initialized = true
     }
-
-    override fun mangaDetailsParse(response: Response): SManga = runBlocking {
-        response.parseAs<Hentai>().toSManga()
+    private fun Elements.emptyOrNull(): Elements? {
+        return this.ifEmpty { null }
     }
 
     override fun getMangaUrl(manga: SManga) = baseUrl + manga.url
 
     override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
+
     override fun chapterListParse(response: Response): List<SChapter> {
-        val hentai = response.parseAs<Hentai>()
+        val doc = response.asJsoup()
 
         return listOf(
             SChapter.create().apply {
                 name = "Chapter"
-                url = "/archive/${hentai.id}/${hentai.slug}"
-                date_upload = hentai.createdAt * 1000
+                url = "/g/${doc.selectFirst("a[href*='/read']")!!.attr("href").substringBefore("/read").substringAfterLast("/")}"
+                date_upload = dateFormat.parse(doc.selectFirst("p:containsOwn(Released)")!!.parent()!!.selectFirst(".text-sm")!!.ownText())!!.time
```

Comment:
Since `date_upload` isn't critical you should avoid using non-null asserts and rather fall back to 0 (or just conditionally set it)